machine_test.go

```diff
@@ -118,13 +118,15 @@ func TestMicroVMExecution(t *testing.T) {
 
 	vmmCtx, vmmCancel := context.WithTimeout(ctx, 30*time.Second)
 	defer vmmCancel()
-	var exitchannel <-chan error
+	exitchannel := make(chan error)
 	go func() {
-		var err error
-		exitchannel, err = m.startVMM(vmmCtx)
+		exitCh, err := m.startVMM(vmmCtx)
 		if err != nil {
+			close(exitchannel)
 			t.Fatalf("Failed to start VMM: %v", err)
 		}
+		exitchannel <- <-exitCh
```

Comment:
Can you give some context here? Why do we need two channels?

Reply:
I'm somewhat confused with this block of code :)

Is this what we want?
```go
vmmCtx, vmmCancel := context.WithTimeout()
...
go func() {

    exitCh, err := m.startVMM(vmmCtx)
    if err != nil {
        t.Fatalf("Failed to start VMM: %v", err)
    } else {
         select {
         case <-vmm.Ctx.Done():
              t.Fatal();
         case <-exitCh:
              OK
        }
    }

}


...

t.Run("TestStartInstance", func(t *testing.T) { testStartInstance(vmmCtx, t, m) })

// We can call this safely, if exit==nil, Fatal is called.
<-exitCh

```

Am I missing something?

Reply:
`t.Fatal` only stops execution of the goroutine, not of the whole test, so that will block forever.

Reply:
Oh. I didn't know what :(